Assets/MixedRealityToolkit.SDK/Features/Utilities/SpatialMeshExporter.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SpatialAwareness.Utilities
+{
+    /// <summary>
+    /// Utility to export current Spatial Mesh Data to OBJ file
+    /// </summary>
+    public static class SpatialMeshExporter
+    {
+        /// <summary>
+        /// Save spatial mesh data for all observers under the current Spatial Awareness system 
+        /// </summary>
+        /// <param name="folderPath">Absolute folder path to place OBJ files</param>
+        /// <param name="consolidate">If true, attempts to consolidate all meshes per Observer into one OBJ file. If false, creates an OBJ file per mesh object on each observer</param>
+        /// <remarks>
+        /// Accessing GameObject/Mesh data will occur as Coroutine on Unity Main thread. May impact performance.
+        /// If folder path does not exist, throws exception
+        /// </remarks>
+        public static async Task Save(string folderPath, bool consolidate = true)
+        {
+            ThrowIfDoesNotExist(folderPath);
```

Comment:
> ThrowIfDoesNotExist(folderPath); [](start = 11, length = 33)

is there a reason why we cannot have folder creation at runtime as an option?

public static async Task Save(string folderPath, bool consolidate = true, bool canCreateFolderIfDoesNotExist = false)

Reply:
Yea it would probably be better to be consistent. I don't' want to make it an argument though. Our functions in general in MRTK are already argument bloated.